### PR TITLE
PROVES-28 Add template based dynamic schema for Elastic8

### PR DESCRIPTION
**Pull request by kehh**
* Remove custom field based config mapping
* Remove many try / catch situations
  * Rather use the built in ElasticSearch functionality of `ignore_missing`
* Don't update application_vars super frequently

### CODE REVIEW

**Comment by bri-gaia** [comment]
LGTM

File: app/lib/Plugins/SearchEngine/Elastic8.php [hunk 1]
```diff
@@ -306,77 +323,18 @@ public function init() {
...
-				}
-			}
-
-			if ( sizeof( $delete_params ) > 0 ) {
-				$this->getClient()->bulk( $delete_params );
-			}
+			$tables = $mapping->getTables();
 		}
-
+		$this->getClient()->indices()->delete( [ 'index' =>  array_map($this->getIndexName, $tables), 'ignore_unavailable' => true ] );
```
Comment (bri-gaia) [inline app/lib/Plugins/SearchEngine/Elastic8.php:336]: `$this->getIndexName` should this be something like `$this->getIndexName($table)` or am I missing something else?

Comment (kehh) [reply app/lib/Plugins/SearchEngine/Elastic8.php:336]: It's an array_map with the first argument being a callable or a string representing a callable.

Comment (kehh) [reply app/lib/Plugins/SearchEngine/Elastic8.php:336]: Changes :
```php
['ca_objects', 'ca_occurrences'...]
// to
['collectiveaccess_ca_objects', 'collectiveaccess_ca_occurrences'...]
```

Comment (bri-gaia) [reply app/lib/Plugins/SearchEngine/Elastic8.php:336]: but there isn't a property called getIndexName? there is a function with that name?

Comment (bri-gaia) [reply app/lib/Plugins/SearchEngine/Elastic8.php:336]: image

Comment (kehh) [reply app/lib/Plugins/SearchEngine/Elastic8.php:336]: Can you replace it with:
```php
$this->getClient()->indices()->delete( [ 'index' =>  array_map('getIndexName', $tables), 'ignore_unavailable' => true ] );
```

**Pull request merged**